### PR TITLE
Replication

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

**Comment by shriramshankar** [comment]
@iamigo The review app is having issues connection to redis. Once that is fixed we can do the load simulation.

**Comment by pallavi2209** [comment]
@shriramshankar @iamigo Please invite me as well you run load simulation.

**Comment by shriramshankar** [comment]
@iamigo @pallavi2209 this PR is ready to be reviewed. 
I thought it was not working in the morning as Postgres was taking a bit longer to log the metrics on the follower and I had also misspelled a config variable.

File: db/utils.js [hunk 1]
```diff
@@ -53,6 +51,99 @@ function dbConfigObjectFromDbURL(dbUrl) {
...
+   * The sequelize object is constructed this way if read-only replicas are
+   * configured.
+   * this usage is of the form new Sequelize('database',
+   * 'username' 'password , options)
+   * the username and password are passed to the constructor through the
+   * replication property of options
+   */
+  seq = new Sequelize(primaryDb.name, null, null, {
+    dialect: env.dialect,
+
```
Comment (iamigo) [inline db/utils.js]: no empty line here

File: db/utils.js [hunk 2]
```diff
@@ -53,6 +51,99 @@ function dbConfigObjectFromDbURL(dbUrl) {
...
+    replication: getDBReplicationObject(primaryDb),
+
```
Comment (iamigo) [inline db/utils.js]: no empty line here

File: db/utils.js [hunk 3]
```diff
@@ -53,6 +51,99 @@ function dbConfigObjectFromDbURL(dbUrl) {
...
+const primaryDb = dbConfigObjectFromDbURL();
+
+// create the sequelize object from the constructor.
+let seq;
+if (conf.readReplicas) {
+
+  /*
+   * The sequelize object is constructed this way if read-only replicas are
+   * configured.
+   * this usage is of the form new Sequelize('database',
```
Comment (iamigo) [inline db/utils.js]: Sentence case, period after sentence

File: db/utils.js [hunk 4]
```diff
@@ -53,6 +51,99 @@ function dbConfigObjectFromDbURL(dbUrl) {
...
+   * 'username' 'password , options)
+   * the username and password are passed to the constructor through the
```
Comment (iamigo) [inline db/utils.js]: Sentence case, period after sentence

**Comment by shriramshankar** [comment]
@iamigo I also do not see USE_READ_ONLY_REPLICAS adding a lot of value. If we add a flag like this, we can also argue that we need toggles to turn on/off other features like IP restriction/ rate limiter. Let me know if you want to add this toggle. I will update the "docs on enterprise-scale" story with information about how to set this up.

**Comment by iamigo** [comment]
OK, no extra env var.

**Pull request merged**